crypto/asn1/a_mbstr.c

```diff
@@ -272,12 +270,15 @@ static int out_utf8(unsigned long value, void *arg)
 static int type_str(unsigned long value, void *arg)
 {
     unsigned long types;
+    const int native = ossl_fromascii(value);
```

Comment:
Using _uint32_t_ and _int32_t_ has the implementation defined cast to signed pitfall but it won't run into size issues.

This will be a more invasive change.